components/error-boundary/error-boundary.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
```

Comment:
Well, it turns out client boundaries don't catch server errors, maybe this functionality can be written using try/catch and async components, but client error boundaries are not useful =)
![Screenshot 2024-11-15 at 1 51 22 PM](https://github.com/user-attachments/assets/c15264b3-d4cc-48ac-8063-718ca8559e12)

See my [chat with v0](https://v0.dev/chat/zspXt5IFSLj) for some ideas explored